crates/ruma-common/src/events/room/encrypted/relation_serde.rs

```diff
@@ -18,8 +18,20 @@ impl<'de> Deserialize<'de> for Relation {
         let ev = EventWithRelatesToJsonRepr::deserialize(deserializer)?;
 
         #[cfg(feature = "unstable-msc3440")]
-        if let Some(RelationJsonRepr::Thread(ThreadJsonRepr { event_id, is_falling_back })) =
-            ev.relates_to.relation
+        if let Some(RelationJsonRepr::ThreadStable(ThreadStableJsonRepr {
```

Comment:
This branch and the other one can be merged if you use `match` instead of `if let` (or maybe `if let` works too with a nested or pattern, but `match` should be more readable).